File: samples_c/s63_palindrome_bytes.c
#include <stdio.h>
#include <string.h>

/* Girdi: tek satir metin.
   Cikti: "EVET" (bayt dizisi kendinin tersiyse) veya "HAYIR".
   NOT: Kok Neden B (string modeli) icin 5. bagimsiz ornek. Karsilastirma
   dogrudan BAYT indeksiyle yapilir (buf[i] vs buf[len-1-i]); cok baytli
   UTF-8 girdide bu, kod noktasi bazinda simetrik olan bir metni bile
   bayt duzeyinde asimetrik gosterebilir - Rust'ta .chars().rev() kullanan
   "dogal" bir ceviri farkli sonuc verir. */
int main(void) {
    char buf[4096];
    if (fgets(buf, sizeof(buf), stdin) == NULL) return 0;
    size_t len = strlen(buf);
    while (len > 0 && (buf[len - 1] == '\n' || buf[len - 1] == '\r')) {
        len--;
    }
    int ok = 1;
    for (size_t i = 0; i < len / 2; i++) {
        if (buf[i] != buf[len - 1 - i]) {
            ok = 0;
            break;
        }
    }
    printf("%s\n", ok ? "EVET" : "HAYIR");
    return 0;
}

File: translations_rust__gemini/s63_palindrome_bytes.rs
use std::io::{self, Read};

fn main() {
    let mut buf = [0u8; 4096];
    let mut read_len = 0;
    let stdin = io::stdin();
    let mut handle = stdin.lock();

    while read_len < 4095 {
        let mut byte = [0u8; 1];
        match handle.read(&mut byte) {
            Ok(1) => {
                buf[read_len] = byte[0];
                read_len += 1;
                if byte[0] == b'\n' {
                    break;
                }
            }
            _ => break,
        }
    }

    if read_len == 0 {
        return;
    }

    let mut len = read_len;
    for i in 0..read_len {
        if buf[i] == 0 {
            len = i;
            break;
        }
    }

    while len > 0 && (buf[len - 1] == b'\n' || buf[len - 1] == b'\r') {
        len -= 1;
    }

    let mut ok = true;
    for i in 0..len / 2 {
        if buf[i] != buf[len - 1 - i] {
            ok = false;
            break;
        }
    }

    if ok {
        println!("EVET");
    } else {
        println!("HAYIR");
    }
}
